### ISSUE TITLE
jsts server vs. plain js server

### ISSUE THREAD

**Issue by james2doyle** [tomv564/LSP#369]
I came to this project to get the PHP LS (which works great) running as well as a JavaScript LS running. It seems like there is a [stale issue on the TypeScript Sublime plugin](https://github.com/Microsoft/TypeScript-Sublime-Plugin/pull/570) for supporting plain JavaScript with the plugin.

I never managed to get that working. So I turned to this project.

Eventually, I did manage to get a "typescript" LS working with plain JavaScript. But I needed to use a [different LS to get that working](https://github.com/theia-ide/typescript-language-server) than the "official" one.

![demo with plain JS](https://cl.ly/140l1A402F01/Screen%20Recording%202018-07-04%20at%2009.13%20PM.gif)

For context, this is a plain `.js` file in a project with no `tsconfig.json` file setup. I did have `@types/lodash` installed through my `package.json`. So it will load the type definitions and display them properly in `.js` files as well as the context of the actual plain JavaScript code.

Once globally installing [theia-ide/typescript-language-server](https://github.com/theia-ide/typescript-language-server), you can then continue on with the following settings:

```json
{
  "js": {
    "command": ["typescript-language-server", "--stdio"],
    "enabled": true,
    "languageId": "javascript",
    "scopes": ["source.js"],
    "syntaxes": ["Packages/JavaScript/JavaScript.sublime-syntax"]
  }
}
```

Basically, this means you can get a much better experience in Sublime with JavaScript using this language server and the LSP plugin.

For general TypeScript usage, the Sublime TypeScript plugin seems to do a fine job. But for plain JavaScript project (not written in `.ts` files) this is excellent.

Essentially, I was wondering if you were interesting in an addition to the docs that outlined this process

**Comment by tomv564** [comment]
Yes, let's add theia-ide's typescript server to the docs please!

I should also add my own (https://github.com/tomv564/lsp-tsserver)

**Comment by james2doyle** [comment]
@tomv564 just checked out your version - it works even better than the theia-ide one!

I can add docs for that then. I will open a new PR for that when I get the chance

**Comment by james2doyle** [comment]
If you could publish your tomv564/lsp-tsserver package on NPM that would help with the instructions

**Comment by predragnikolic** [comment]
You can find Tom's server on npm [here]
(https://www.npmjs.com/package/lsp-tsserver). :-)
